numpy/ma/tests/test_core.py

```diff
@@ -492,78 +480,81 @@ def test_setitem_no_warning(self):
-        n = [0, 0, 1, 0, 0]
-        m = make_mask(n)
-        m2 = make_mask(m)
-        assert_(m is m2)
-        m3 = make_mask(m, copy=True)
-        assert_(m is not m3)
-
-        x1 = np.arange(5)
-        y1 = array(x1, mask=m)
-        assert_equal(y1._data.__array_interface__, x1.__array_interface__)
-        assert_(allequal(x1, y1.data))
-        assert_equal(y1._mask.__array_interface__, m.__array_interface__)
-
-        y1a = array(y1)
-        # Default for masked array is not to copy; see gh-10318.
-        assert_(y1a._data.__array_interface__ ==
-                        y1._data.__array_interface__)
-        assert_(y1a._mask.__array_interface__ ==
-                        y1._mask.__array_interface__)
-
-        y2 = array(x1, mask=m3)
-        assert_(y2._data.__array_interface__ == x1.__array_interface__)
-        assert_(y2._mask.__array_interface__ == m3.__array_interface__)
-        assert_(y2[2] is masked)
-        y2[2] = 9
-        assert_(y2[2] is not masked)
-        assert_(y2._mask.__array_interface__ == m3.__array_interface__)
-        assert_(allequal(y2.mask, 0))
-
-        y2a = array(x1, mask=m, copy=1)
-        assert_(y2a._data.__array_interface__ != x1.__array_interface__)
-        #assert_( y2a._mask is not m)
-        assert_(y2a._mask.__array_interface__ != m.__array_interface__)
-        assert_(y2a[2] is masked)
-        y2a[2] = 9
-        assert_(y2a[2] is not masked)
-        #assert_( y2a._mask is not m)
-        assert_(y2a._mask.__array_interface__ != m.__array_interface__)
-        assert_(allequal(y2a.mask, 0))
-
-        y3 = array(x1 * 1.0, mask=m)
-        assert_(filled(y3).dtype is (x1 * 1.0).dtype)
-
-        x4 = arange(4)
-        x4[2] = masked
-        y4 = resize(x4, (8,))
-        assert_equal(concatenate([x4, x4]), y4)
-        assert_equal(getmask(y4), [0, 0, 1, 0, 0, 0, 1, 0])
-        y5 = repeat(x4, (2, 2, 2, 2), axis=0)
-        assert_equal(y5, [0, 0, 1, 1, 2, 2, 3, 3])
-        y6 = repeat(x4, 2, axis=0)
-        assert_equal(y5, y6)
-        y7 = x4.repeat((2, 2, 2, 2), axis=0)
-        assert_equal(y5, y7)
-        y8 = x4.repeat(2, 0)
-        assert_equal(y5, y8)
-
-        y9 = x4.copy()
-        assert_equal(y9._data, x4._data)
-        assert_equal(y9._mask, x4._mask)
-
-        x = masked_array([1, 2, 3], mask=[0, 1, 0])
-        # Copy is False by default
-        y = masked_array(x)
-        assert_equal(y._data.ctypes.data, x._data.ctypes.data)
-        assert_equal(y._mask.ctypes.data, x._mask.ctypes.data)
-        y = masked_array(x, copy=True)
-        assert_not_equal(y._data.ctypes.data, x._data.ctypes.data)
-        assert_not_equal(y._mask.ctypes.data, x._mask.ctypes.data)
+        msg = "setting an item on a masked array which has a shared mask will not copy"
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                'ignore', msg, numpy.ma.core.MaskedArrayFutureWarning)
```

Comment:
why not use `@pytest.mark.filterwarnings` is it not thread-safe enough?  That way you don't have to change the indentation (and the extra level is annoying).